src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -243,62 +244,59 @@ internal static partial class Library
                     targetFunction: Exp)
             },
             {
-                BuiltinFunctionsCore.IsBlank,
+                BuiltinFunctionsCore.Filter,
                 StandardErrorHandling<FormulaValue>(
                     expandArguments: NoArgExpansion,
                     replaceBlankValues: DoNotReplaceBlank,
-                    checkRuntimeTypes: DeferRuntimeTypeChecking,
+                    checkRuntimeTypes: ExactSequence(
+                        ExactValueTypeOrBlank<TableValue>,
+                        ExactValueTypeOrBlank<LambdaFormulaValue>),
                     checkRuntimeValues: DeferRuntimeValueChecking,
-                    returnBehavior: ReturnBehavior.AlwaysEvaluateAndReturnResult,
-                    targetFunction: IsBlank)
-            },
-            {
-                BuiltinFunctionsCore.IsError,
-                IsError
+                    returnBehavior: ReturnBehavior.ReturnBlankIfAnyArgIsBlank,
+                    targetFunction: FilterTable)
             },
             {
-                BuiltinFunctionsCore.IsNumeric,
+                BuiltinFunctionsCore.Find,
                 StandardErrorHandling<FormulaValue>(
-                    expandArguments: NoArgExpansion,
-                    replaceBlankValues: DoNotReplaceBlank,
-                    checkRuntimeTypes: DeferRuntimeTypeChecking,
+                    expandArguments: InsertDefaultValues(outputArgsCount: 3, fillWith: new NumberValue(IRContext.NotInSource(FormulaType.Number), 1)),
+                    replaceBlankValues: ReplaceBlankWith(
+                        new StringValue(IRContext.NotInSource(FormulaType.String), string.Empty),
+                        new StringValue(IRContext.NotInSource(FormulaType.String), string.Empty),
+                        new BlankValue(IRContext.NotInSource(FormulaType.Blank))),
+                    checkRuntimeTypes: ExactSequence(
+                        ExactValueType<StringValue>,
+                        ExactValueType<StringValue>,
+                        ExactValueTypeOrBlank<NumberValue>),
                     checkRuntimeValues: DeferRuntimeValueChecking,
```

Comment:
This function accepts a numeric value as input, so we need to check that it is not infinite. There is a runtime value check that does this already.

Reply:
There is additional validation that needs to be done on startIndex that can't be done using runtime-value-check infra (logic needs to look at twi args at once). I added infinities check and I think keeping all validation in one place (either runtime-value-check xor func impl) is better than splitting. Open for discussion